modules/affile/tests/test_wildcard_file_reader.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2018 Balabit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include <criterion/criterion.h>
+#include <glib/gstdio.h>
+#include <unistd.h>
+#include "apphook.h"
+#include "wildcard-file-reader.h"
+#include "logpipe.h"
+#include "iv.h"
+
+#define TEST_FILE_NAME "TEST_FILE"
+
+
+typedef struct _TestFileStateEvent
+{
+  FileStateEvent super;
+  gboolean deleted_eof_called;
+  gboolean finished_called;
+} TestFileStateEvent;
+
+static void
+_eof(FileReader *reader, gpointer user_data)
+{
+  TestFileStateEvent *test = (TestFileStateEvent *) user_data;
+  test->deleted_eof_called = TRUE;
+}
+
+static void
+_finished(FileReader *reader, gpointer user_data)
+{
+  TestFileStateEvent *test = (TestFileStateEvent *) user_data;
+  test->finished_called = TRUE;
+}
+
+TestFileStateEvent *
+test_deleted_file_state_event_new(void)
+{
+  TestFileStateEvent *self = g_new0(TestFileStateEvent, 1);
+  self->super.user_data = self;
+  self->super.deleted_file_eof = _eof;
+  self->super.deleted_file_finised = _finished;
+  return self;
+}
+
+TestFileStateEvent *test_event;
+WildcardFileReader *reader;
+LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
+
+/* MOCKS */
+
+gboolean file_reader_init_method(LogPipe *s)
+{
+  return TRUE;
+}
+
+gboolean file_reader_deinit_method(LogPipe *s)
+{
+  return TRUE;
+}
+
+void file_reader_queue_method(LogPipe *s, LogMessage *msg, const LogPathOptions *po)
+{
+  return;
+}
+
+void file_reader_notify_method(LogPipe *s, gint notify_code, gpointer user_data)
+{
+  return;
+}
+
+int iv_task_registered(const struct iv_task *_t)
```

Comment:
Just as a reminder, this will not compile with below libivykis 0.38.

Reply:
I've fixed it